src/server/saved_objects/service/create_saved_objects_service.js

```diff
@@ -30,14 +30,17 @@ import { SavedObjectsClient } from './saved_objects_client';
 // import / export from the Kibana UI. Import / export functionality needs to apply migrations to documents.
 // Eventually, we hope to build a first-class import / export API, at which point, we can
 // remove the migrator from the saved objects client and leave only document validation here.
-export function createSavedObjectsService(server, schema, serializer, migrator) {
+export function createSavedObjectsService(server, schema, serializer, migrator, extraTypes) {
   const mappings = server.getKibanaIndexMappingsDsl();
+  const allTypes = Object.keys(getRootPropertiesObjects(mappings));
+  const allowedTypes = [...new Set((extraTypes || []).concat(allTypes.filter(type => !schema.isHiddenType(type))))];
```

Comment:
`extraTypes` is exactly that it *does allow* for the repository to be configured for additional types that may not be in the mappings. This is intentional and part of the design in order to be able to expose the hidden types. I suppose this list could expose the hidden types differently  and verify that they are part of the mapping. Then it would prevent a repository from being created with truly `unknown types` from this method, but I wanted to keep that option open in case you need to write a plugin to work on old types that are no longer used or other saved objects that happen to be in your .kibana index.

As far as the duplicates go it's being put into a `Set` which will de-dup the list so that isn't a concern. 

If we only allowed mapped types to be created here then we might call the variable `includeTypes` instead, though that makes more sense as a method than a parameter name `includedTypes` is a little bit off meaning-wise as that suggests the service will only work on whats in the list, hence I went with `extraTypes`. I had thought about using the wording `expose` though that does not apply to repository. Since the repository and this method do not create or control routes there is no `exposing` to be done. And I would not want to give the impression that adding to the list allowed those types to be manipulated via the REST api.

Perhaps an alternative might be `additionalTypes` but I keep coming back to `extraTypes` hence why that was the name I chose. That is essentially what they are except that they could collide with already defined types but their merged anyway and in that case the type wasn't extra. Another option is `otherTypes`.

Reply:
> `extraTypes` is exactly that it _does allow_ for the repository to be configured for additional types that may not be in the mappings

If the type isn't part of the Kibana index mapping, how is the repository supposed to interact with objects of these types? It won't be able to create them, since that would violate Kibana's strict index mapping.

> As far as the duplicates go it's being put into a `Set` which will de-dup the list so that isn't a concern.

hmm, I'm not seeing that. It looks to me like `extraTypes` is part of a set, but that set put into an array and then concat'ed with the list of visible types. If `extraTypes` contains, for example, `index-pattern`, then `allowedTypes` will contain `index-pattern` twice: once from `extraTypes`, and once from `allTypes`. I know specifying visible types isn't the intent of `extraTypes`, but it's something that is technically allowed here.

Reply:
You guys are right about the types without mappings not working. I can throw an exception if a `extraType` does not have a mapping. 

@legrego That the entire concatenated list of `extraTypes` with `allTypes` is passed first to the Set c'tor and then destructed into an array. I'll rework to make that clear.

Reply:
Also `includedHiddenTypes` or `additionalTypes` or `otherTypes` might make sense now. Since by default the only types that would change the Repository from the default one with all types would be those types that are hidden.